website/views.py

```diff
@@ -277,7 +278,10 @@ def resolve_guid(guid, suffix=None):
                     # the routing scheme - if a preprint is not published, only
                     # admins should be able to know it exists.
                     auth = Auth.from_kwargs(request.args.to_dict(), {})
-                    if not referent.node.has_permission(auth.user, permissions.ADMIN):
+                    group_helper = GroupHelper(referent.provider)
+                    admin_group = group_helper.get_group('admin')
+                    mod_group = group_helper.get_group('moderator')
+                    if not (referent.node.has_permission(auth.user, permissions.ADMIN) or mod_group.user_set.filter(username=auth.user).exists() or admin_group.user_set.filter(username=auth.user).exists()):
```

Comment:
Isn't `auth.user` here an `OSFUser` object? I'm assuming Django is doing some magic to let a `str == object` comparison happen.

Could you do a little benchmarking to see if:
1. this query becomes faster when using `filter(id=auth.user.id)`
2. whether or not it's more efficient to do 
`(mod_group.user_set | admin_group.user_set).filter(id=auth.user.id)`

I can assist with benchmarking IRL if you need